Source/SuperLinq/SortedMerge.cs

```diff
@@ -355,48 +355,124 @@ public static IEnumerable<TSource> SortedMerge<TSource>(this IEnumerable<TSource
 		// Private implementation method that performs a merge of multiple, ordered sequences using
 		// a precedence function which encodes order-sensitive comparison logic based on the caller's arguments.
 		//
-		// The algorithm employed in this implementation is not necessarily the most optimal way to merge
-		// two sequences. A swap-compare version would probably be somewhat more efficient - but at the
-		// expense of considerably more complexity. One possible optimization would be to detect that only
-		// a single sequence remains (all other being consumed) and break out of the main while-loop and
-		// simply yield the items that are part of the final sequence.
+		// Where available, PriorityQueue is used to maintain the remaining enumerators ordered by the
+		// value of the Current property.
 		//
-		// The algorithm used here will perform N*(K1+K2+...Kn-1) comparisons, where <c>N => otherSequences.Count()+1.</c>
+		// Otherwise, a sorted array is used, with BinarySearch finding the re-insert location.
 
 		static IEnumerable<TSource> Impl(IEnumerable<IEnumerable<TSource>> sequences, Func<TSource, TKey> keySelector, IComparer<TKey> comparer)
 		{
-			using var list = new EnumeratorList<TSource>(sequences);
+			var enumerators = new List<IEnumerator<TSource>>();
 
-			// prime all of the iterators by advancing them to their first element (if any)
-			for (var i = 0; list.MoveNext(i); i++)
-			{ }
-
-			// while all iterators have not yet been consumed...
-			while (list.Any())
+			try
 			{
-				var nextIndex = 0;
-				var nextValue = list.Current(0);
-				var nextKey = keySelector(nextValue);
+				// Ensure we dispose first N enumerators if N+1 throws 
+				foreach (var sequence in sequences)
+				{
+					var e = sequence.GetEnumerator();
+					if (e.MoveNext())
+					{
+						enumerators.Add(e);
+					}
+					else
+					{
+						e.Dispose();
+					}
+				}
+
+#if NET6_0_OR_GREATER
+				var queue = new PriorityQueue<IEnumerator<TSource>, TKey>(
+					enumerators.Select(x => (x, keySelector(x.Current))),
+					comparer);
+
+#pragma warning disable CA2000 // e will be disposed via enumerators list
+				while (queue.TryDequeue(out var e, out var _))
+#pragma warning restore CA2000 // Dispose objects before losing scope
+				{
+					yield return e.Current;
```

Comment:
```suggestion
				while (queue.TryDequeue(out var e, out var current))
#pragma warning restore CA2000 // Dispose objects before losing scope
				{
					yield return current;
```

Reply:
The second output isn't `e.Current`, it's the key, i.e.`keySelector(e.Current)`.